src/fmtargs.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Redis Contributors.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD 3-Clause
+ *
+ * To make it easier to map each part of the format string with each argument,
+ * this file provides a way to write
+ *
+ *     printf("a = %s, b = %s, c= %s\n",
+ *            arg1, arg2, arg3);
+ *
+ * as
+ *
+ *     printf(FMTARGS("a = %s, ", arg1,
+ *                    "b = %s, ", arg2,
+ *                    "c = %s\n", arg3));
+ */
+#ifndef FMTARGS_H
+#define FMTARGS_H
+
+#define NARG(...)  NARG_I(__VA_ARGS__,RSEQ_N())
+#define NARG_I(...) ARG_N(__VA_ARGS__)
```

Comment:
to make it easier to read, i suggest adding a comment above each of these macros (or macro groups)